dref/serializers.py

```diff
@@ -0,0 +1,217 @@
+    created_by_details = UserNameSerializer(source='created_by', read_only=True)
+    users_details = UserNameSerializer(source='users', many=True, read_only=True)
+    budget_file_details = DrefFileSerializer(source='budget_file', read_only=True)
+    cover_image_details = DrefFileSerializer(source='cover_image', read_only=True)
+    disaster_type_details = DisasterTypeSerializer(source='disaster_type', read_only=True)
+
+    class Meta:
+        model = Dref
+        fields = '__all__'
+        read_only_fields = ('modified_by', 'created_by', 'budget_file_preview')
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        disability_people_per = data.get('disability_people_per', '')
+        if disability_people_per and len(disability_people_per.split('.')[0]) == 3:
+            data['disability_people_per'] = disability_people_per.split('.')[0]
+        people_per_urban = data.get('people_per_urban', '')
+        people_per_local = data.get('people_per_local', '')
+        if people_per_urban and len(people_per_urban.split('.')[0]) == 3:
+            data['people_per_urban'] = people_per_urban.split('.')[0]
+        if people_per_local and len(people_per_local.split('.')[0]) == 3:
+            data['people_per_local'] = people_per_local.split('.')[0]
+        return data
+
+    def validate(self, data):
+        event_date = data.get('event_date', None)
+        if event_date and data['type_of_onset'] not in [Dref.OnsetType.SLOW, Dref.OnsetType.SUDDEN]:
+            raise serializers.ValidationError({
+                'event_date': ugettext('Cannot add event_date if onset type not in {} or {}')
+                .format(Dref.OnsetType.SLOW.label, Dref.OnsetType.SUDDEN.label)
+            })
+        return data
+
+    def validate_images(self, images):
+        # Validate if images are provided
+        if images:
+            # Fetch current images
+            current_images_map = {
+                image.pk: image.file
+                for image in (self.instance and self.instance.images.all()) or []
+            }
+            for image in images:
+                # Ignore for current images (if file is not changed)
+                if image.id is not None and image.id in current_images_map and image.file == current_images_map[image.id]:
+                    continue
+                elif image.created_by != self.context['request'].user:
+                    raise serializers.ValidationError(
+                        ugettext('Only image owner can attach image(id: %s)' % image.id)
+                    )
+        # Don't allow images more than MAX_NUMBER_OF_IMAGES
+        if len(images) > self.MAX_NUMBER_OF_IMAGES:
+            raise serializers.ValidationError(
+                ugettext('Can add utmost %s images' % self.MAX_NUMBER_OF_IMAGES)
+            )
+        return images
+
+    def validate_extension(self, filename):
```

Comment:
Is this used by other than validate_budget_file or by extension field? If not let's scope this inside validate_budget_file

Reply:
moved